engine/option.go

```diff
@@ -0,0 +1,28 @@
+package engine
+
+import "fmt"
+
+// engineConfig contains all options for engineConfiguring Engine.
+type engineConfig struct {
+	minerID string
+}
+
+type Option func(*engineConfig) error
+
+// apply applies the given options to this engineConfig.
+func (c *engineConfig) apply(opts []Option) error {
+	for i, opt := range opts {
+		if err := opt(c); err != nil {
+			return fmt.Errorf("option %d failed: %s", i, err)
+		}
+	}
+	return nil
+}
+
+// WithMinerID sets the miner ID to include in the pubsub announcement.
+func WithMinerID(minerID string) Option {
```

Comment:
even here i'd have index provider interface be neutral:
`WithExtraGossipData(extraData string) Option` or similar

Reply:
OK, hang on...

Reply:
Prefer `string` or `[]byte`?

Reply:
I would say whatever it is in legs